src/3d/qgscameracontroller.cpp

```diff
@@ -432,6 +485,48 @@ void QgsCameraController::onKeyPressed( Qt3DInput::QKeyEvent *event )
   }
 }
 
+void QgsCameraController::onKeyPressedFPSNavigation( Qt3DInput::QKeyEvent *event )
+{
+  QVector3D cameraUp = mCamera->upVector().normalized();
+  QVector3D cameraFront = ( QVector3D( mCameraPose.centerPoint().x(), mCameraPose.centerPoint().y(), mCameraPose.centerPoint().z() ) - mCamera->position() ).normalized();
+  QVector3D cameraLeft = QVector3D::crossProduct( cameraUp, cameraFront );
+
+  QVector3D cameraPosDiff( 0.0f, 0.0f, 0.0f );
+  float movementSpeed = 5.0f;
```

Comment:
This would probably better be configurable in settings. Different scale projects will require different speeds!